File: smells/unclear/chafing/chafing.js
// Subject under test
function pricingForCode (code) {
  var firstFactor = 0
  if (code[0] === 'A') firstFactor = 65
  if (code[0] === '7') firstFactor = 55
  if (code[0] === '(') firstFactor = 40

  var secondFactor = 0
  if (code.length === 3) secondFactor = 21
  if (code.length === 5) secondFactor = 19
  if (code.length === 8) secondFactor = 16

  return firstFactor * secondFactor
}

// Test
var generateCode = require('../../../support/js/generate-code')
module.exports = {
  codeOneIsCorrect: function () {
    var code = generateCode.one()

    var result = pricingForCode(code)

    assert.codePricing(result, code)
  },
  codeTwoIsCorrect: function () {
    var code = generateCode.two()

    var result = pricingForCode(code)

    assert.codePricing(result, code)
  },
  codeThreeIsCorrect: function () {
    var code = generateCode.three()

    var result = pricingForCode(code)

    assert.codePricing(result, code)
  }
}

File: smells/unclear/chafing/chafing.rb
# Subject under test
def pricing_for_code(code)
  first_factor = 0
  first_factor = 65 if code[0] == 'A'
  first_factor = 55 if code[0] == '7'
  first_factor = 40 if code[0] == '('

  second_factor = 0
  second_factor = 21 if code.size == 3
  second_factor = 19 if code.size == 5
  second_factor = 16 if code.size == 8

  return first_factor * second_factor
end

# Test
require_relative "../../../support/ruby/generate_code"

class Chafing < SmellTest
  def test_code_one_is_correct
    code = GenerateCode.one()

    result = pricing_for_code(code)

    assert_code_pricing code, result
  end

  def test_code_two_is_correct
    code = GenerateCode.two()

    result = pricing_for_code(code)

    assert_code_pricing code, result
  end

  def test_code_three_is_correct
    code = GenerateCode.three()

    result = pricing_for_code(code)

    assert_code_pricing code, result
  end
end

